manifests/repository.pp

```diff
@@ -43,30 +84,47 @@
       yumrepo { 'bareos':
         name     => 'bareos',
         descr    => 'Bareos Repository',
+        username => $yum_username,
+        password => $yum_password,
         baseurl  => $location,
         gpgcheck => '1',
         gpgkey   => "${location}/repodata/repomd.xml.key",
         priority => '1',
       }
     }
     /(?i:debian|ubuntu)/: {
+      if $subscription {
+        $url = "${scheme}${username}:${password}@${address}"
+      } else {
+        $url = "${scheme}${address}"
+      }
       if $os  == 'Ubuntu' {
-        unless $osrelease in ['12.04', '14.04', '16.04', '18.04'] {
+        unless $osrelease in ['12.04', '14.04', '16.04', '18.04', '20.04'] {
```

Comment:
we currently don't list Ubuntu 20 as supported in the metadata.json. Can you add it to the metadata.json in another PR?

Reply:
I removed it from this PR and I'll open a new one for ubuntu 20.04 support